joint_state_publisher/joint_state_publisher/joint_state_publisher.py

```diff
@@ -161,10 +204,16 @@ def configure_robot(self, description):
         self.free_joints = {}
         self.joint_list = [] # for maintaining the original order of the joints
 
-        if robot.getElementsByTagName('COLLADA'):
+        # Get root tag to parse file format
+        root = robot.documentElement
+        if root.tagName == 'sdf':
+            self.init_sdf(robot)
+        elif root.tagName == 'COLLADA':
             self.init_collada(robot)
-        else:
+        elif root.tagName == 'urdf':
             self.init_urdf(robot)
+        else:
+            self.get_logger().error('Cannot parse file of type %s' % root.tagName)
```

Comment:
If we are going to error here, we should also `return`.

Reply:
Also, I can't put a comment on the code below due to GitHub UI limitations, but in `__init__` we should change the name of the parameter from `urdf_file` to just `description_file` or something like that.